Conversion of Dozor bpmn throws validation error

I believe I have the newest version of the bpmn2stamp converter:
```
..bpmn-convertor-examples$ md5sum .target/bpmn2stamp.jar 
812aa0cf100edf3d22d3059f92720bec  .target/bpmn2stamp.jar
```

But when running:
`..bpmn-convertor-examples/processes/ucl$ ./bin/process-bpmn.sh ./bpmn/dozor-nad-provozovateli-letist.bpmn`


It throws error:
```
[main] INFO ConverterMappingService - Actor mapping file _wW8TAFviEeqIT4J6BYWACw.xml was parsed as a normal mapping file
[main] ERROR RdfRepositoryWriter - Could not save ontology, rollback...
[main] ERROR RdfRepositoryWriter - Constraints violation error: At least 1 values of attribute SequenceFlow.has_targetRef expected in instance http://onto.fel.cvut.cz/ontologies/processes/ucl/dozor-nad-provozovateli-letist-bpmn/_qteqkhhreeqzdqdldgvjca, but got only 0
Exception in thread "main" cz.cvut.kbss.jopa.exceptions.RollbackException: cz.cvut.kbss.jopa.exceptions.CardinalityConstraintViolatedException: At least 1 values of attribute SequenceFlow.has_targetRef expected in instance http://onto.fel.cvut.cz/ontologies/processes/ucl/dozor-nad-provozovateli-letist-bpmn/_qteqkhhreeqzdqdldgvjca, but got only 0
        at cz.cvut.kbss.jopa.transactions.EntityTransactionImpl.commit(EntityTransactionImpl.java:60)
        at cz.cvut.kbss.bpmn2stamp.converter.persistance.RdfRepositoryWriter.write(RdfRepositoryWriter.java:82)
        at cz.cvut.kbss.bpmn2stamp.console.Bpmn2StampConverterService.saveToRdf(Bpmn2StampConverterService.java:116)
        at cz.cvut.kbss.bpmn2stamp.console.Bpmn2StampConverterService.saveToRdf(Bpmn2StampConverterService.java:120)
        at cz.cvut.kbss.bpmn2stamp.console.Bpmn2StampConverterService.convertToBbo(Bpmn2StampConverterService.java:38)
        at cz.cvut.kbss.bpmn2stamp.console.Bpmn2StampConverterService.doConversionWithUsingReasoner(Bpmn2StampConverterService.java:87)
        at cz.cvut.kbss.bpmn2stamp.console.Bpmn2StampConverterService.convertToStampAndBbo(Bpmn2StampConverterService.java:67)
        at cz.cvut.kbss.bpmn2stamp.console.ConsoleRunner.run(ConsoleRunner.java:104)
        at cz.cvut.kbss.bpmn2stamp.console.ConsoleRunner.main(ConsoleRunner.java:67)
Caused by: cz.cvut.kbss.jopa.exceptions.CardinalityConstraintViolatedException: At least 1 values of attribute SequenceFlow.has_targetRef expected in instance http://onto.fel.cvut.cz/ontologies/processes/ucl/dozor-nad-provozovateli-letist-bpmn/_qteqkhhreeqzdqdldgvjca, but got only 0
        at cz.cvut.kbss.jopa.sessions.validator.CardinalityConstraintsValidator.validateParticipationConstraint(CardinalityConstraintsValidator.java:89)
        at cz.cvut.kbss.jopa.sessions.validator.CardinalityConstraintsValidator.validate(CardinalityConstraintsValidator.java:66)
        at cz.cvut.kbss.jopa.sessions.validator.GeneralIntegrityConstraintsValidator.validate(GeneralIntegrityConstraintsValidator.java:33)
        at cz.cvut.kbss.jopa.sessions.validator.IntegrityConstraintsValidator.lambda$validate$2(IntegrityConstraintsValidator.java:62)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
        at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
        at java.base/java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1621)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
        at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
        at cz.cvut.kbss.jopa.sessions.validator.IntegrityConstraintsValidator.validate(IntegrityConstraintsValidator.java:60)
        at cz.cvut.kbss.jopa.sessions.UnitOfWorkImpl.validateIntegrityConstraints(UnitOfWorkImpl.java:351)
        at cz.cvut.kbss.jopa.sessions.UnitOfWorkImpl.commitToOntology(UnitOfWorkImpl.java:344)
        at cz.cvut.kbss.jopa.sessions.UnitOfWorkImpl.commitUnitOfWork(UnitOfWorkImpl.java:314)
        at cz.cvut.kbss.jopa.sessions.UnitOfWorkImpl.commit(UnitOfWorkImpl.java:296)
        at cz.cvut.kbss.jopa.transactions.EntityTransactionImpl.commit(EntityTransactionImpl.java:57)
        ... 8 more
```

--- a/bpmn2stamp-converter/src/main/java/cz/cvut/kbss/bpmn2stamp/converter/mapper/bpmn2bbo/MapstructBpmn2BboMapper.java
+++ b/bpmn2stamp-converter/src/main/java/cz/cvut/kbss/bpmn2stamp/converter/mapper/bpmn2bbo/MapstructBpmn2BboMapper.java
@@ -93,6 +93,9 @@ public abstract class MapstructBpmn2BboMapper extends OntologyMapstructMapper<TD
                 mappingContext = new MappingContext(tProcess.getId(), tProcess.getName());
                 
                 Process process = processToProcess(tProcess);
+
+                mappingContext.setProcess(process);
+                
                 result.getProcesses().put(tProcess.getId(), process);
                 for (JAXBElement<? extends TFlowElement> flow : tProcess.getFlowElement()) {
                     TFlowElement flowElement = flow.getValue();
@@ -100,12 +103,6 @@ public abstract class MapstructBpmn2BboMapper extends OntologyMapstructMapper<TD
                     if (resFlowElement == null)
                         continue;
                     result.getFlowElements().put(resFlowElement.getId(), resFlowElement);
-                    getAfterMapping().add(() -> {
-                        if (process.getHas_flowElements() == null) {
-                            process.setHas_flowElements(new HashSet<>());
-                        }
-                        process.getHas_flowElements().add(resFlowElement);
-                    });
                 }
             }
         }
@@ -167,16 +164,25 @@ public abstract class MapstructBpmn2BboMapper extends OntologyMapstructMapper<TD
         getAfterMapping().add(() -> {
             TBaseElement targetRef = (TBaseElement) sequenceFlow.getTargetRef();
             String targetRefTargetId = sourceToTargetIds.get(targetRef.getId());
-            if (getMappedObjectsById().get(targetRefTargetId) != null)
-                normalSequenceFlowResult.setHas_targetRef(Sets.newHashSet(
-                        getMappedObjectsById().get(targetRefTargetId))
-                );
+            Thing targetRefTargetThing = getMappedObjectsById().get(targetRefTargetId);
+            
             TBaseElement sourceRef = (TBaseElement) sequenceFlow.getSourceRef();
             String sourceRefTargetId = sourceToTargetIds.get(sourceRef.getId());
-            if (getMappedObjectsById().get(sourceRefTargetId) != null)
-                normalSequenceFlowResult.setHas_sourceRef(
-                        (FlowNode) getMappedObjectsById().get(sourceRefTargetId)
-                );
+            Thing sourceRefTargetThing = getMappedObjectsById().get(sourceRefTargetId);
+            
+            if (targetRefTargetThing == null || sourceRefTargetThing == null) {
+                getMappedObjectsById().remove(getId(normalSequenceFlowResult));
+                return;
+            }
+            
+            normalSequenceFlowResult.setHas_targetRef(Sets.newHashSet(targetRefTargetThing));
+            normalSequenceFlowResult.setHas_sourceRef((FlowNode) sourceRefTargetThing);
+
+            Process process = mappingContext.getProcess();
+            if (process.getHas_flowElements() == null) {
+                process.setHas_flowElements(new HashSet<>());
+            }
+            process.getHas_flowElements().add(normalSequenceFlowResult);
         });
     }
 
@@ -344,6 +350,7 @@ public abstract class MapstructBpmn2BboMapper extends OntologyMapstructMapper<TD
     static class MappingContext {
         private String poolId;
         private String poolName;
+        private Process process;
 
         public MappingContext(String poolId, String poolName) {
             this.poolId = poolId;
@@ -364,6 +371,14 @@ public abstract class MapstructBpmn2BboMapper extends OntologyMapstructMapper<TD
 
         public void setPoolName(String poolName) {
             this.poolName = poolName;
+        }
+
+        public Process getProcess() {
+            return process;
+        }
+
+        public void setProcess(Process process) {
+            this.process = process;
         }
     }
 

